schema/crdb/dbinit.sql

```diff
@@ -4992,6 +4992,383 @@ CREATE UNIQUE INDEX IF NOT EXISTS one_record_per_volume_resource_usage on omicro
+    schema_version STRING(64) NOT NULL,
+
+    PRIMARY KEY (rx_id, glob)
+);
+
+-- Look up all event class globs for a webhook receiver.
+CREATE INDEX IF NOT EXISTS lookup_webhook_event_globs_for_rx
+ON omicron.public.webhook_rx_event_glob (
+    rx_id
+);
+
+CREATE INDEX IF NOT EXISTS lookup_webhook_event_globs_by_schema_version
+ON omicron.public.webhook_rx_event_glob (schema_version);
+
+CREATE TABLE IF NOT EXISTS omicron.public.webhook_rx_subscription (
+    -- UUID of the webhook receiver (foreign key into
+    -- `omicron.public.webhook_rx`)
+    rx_id UUID NOT NULL,
+    -- An event class to which the receiver is subscribed.
+    event_class omicron.public.webhook_event_class NOT NULL,
+    -- If this subscription is a concrete instantiation of a glob pattern, the
+    -- value of the glob that created it (and, a foreign key into
+    -- `webhook_rx_event_glob`). If the receiver is subscribed to this exact
+    -- event class, then this is NULL.
+    --
+    -- This is used when deleting a glob subscription, as it is necessary to
+    -- delete any concrete subscriptions to individual event classes matching
+    -- that glob.
+    glob STRING(512),
+
+    time_created TIMESTAMPTZ NOT NULL,
+
+    PRIMARY KEY (rx_id, event_class)
+);
+
+-- Look up all webhook receivers subscribed to an event class. This is used by
+-- the dispatcher to determine who is interested in a particular event.
+CREATE INDEX IF NOT EXISTS lookup_webhook_rxs_for_event_class
+ON omicron.public.webhook_rx_subscription (
+    event_class
+);
+
+-- Look up all exact event class subscriptions for a receiver.
+--
+-- This is used when generating a view of all user-provided original
+-- subscriptions provided for a receiver. That list is generated by looking up
+-- all exact event class subscriptions for the receiver ID in this table,
+-- combined with the list of all globs in the `webhook_rx_event_glob` table.
+CREATE INDEX IF NOT EXISTS lookup_exact_subscriptions_for_webhook_rx
+on omicron.public.webhook_rx_subscription (
+    rx_id
+) WHERE glob IS NULL;
+
+/*
+ * Webhook event message queue.
+ */
+
+CREATE TABLE IF NOT EXISTS omicron.public.webhook_event (
+    id UUID PRIMARY KEY,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL,
+    -- The class of event that this is.
+    event_class omicron.public.webhook_event_class NOT NULL,
+    -- Actual event data. The structure of this depends on the event class.
+    event JSONB NOT NULL,
+
+    -- Set when dispatch entries have been created for this event.
+    time_dispatched TIMESTAMPTZ,
+    -- The number of receivers that this event was dispatched to.
+    num_dispatched INT8 NOT NULL,
+
+    CONSTRAINT time_dispatched_set_if_dispatched CHECK (
+        (num_dispatched = 0) OR (time_dispatched IS NOT NULL)
+    ),
+
+    CONSTRAINT num_dispatched_is_positive CHECK (
+        (num_dispatched >= 0)
+    )
+);
+
+-- Singleton probe event
+INSERT INTO omicron.public.webhook_event (
```

Comment:
Our schema-change tests (like, the ones to validate that new deployments match old deployments) will definitely help you (and hopefully did help you??) with ensuring that data-definition language statements (new indices, tables, etc) match between old/new formats.

However, they won't actually help catch **data manipulation language** changes (e.g., "insert into this table") are consistent between deployments.

Since you're adding some of this data, as *part* of the schema change, I recommend checking out `nexus/tests/integration_tests/schema.rs`'s `validate_data_migration` test.

You should be able to query for this row from the `webhook_event` table, and confirm it's there. The `after_125_0_0` migration is doing something very similar.

Reply:
Done in 1ed42a1dde5efa42723b52f661d58f276278950b!